src/v2/Components/FlashBanner/EmailConfirmationCTA.tsx

```diff
@@ -0,0 +1,78 @@
+import React from "react"
+import { commitMutation, graphql } from "react-relay"
+import { Button, Sans } from "@artsy/palette"
+import { BannerContent } from "v2/Components/FlashBanner/types"
+import { EmailConfirmationCTAMutationResponse } from "v2/__generated__/EmailConfirmationCTAMutation.graphql"
+
+export const EmailConfirmationCTA: BannerContent = ({ environment }) => {
+  const [afterSubmitMessage, setAfterSubmitMessage] = React.useState<string>(
+    null
+  )
+
+  const handleSubmit = async () => {
+    console.log(environment)
+    return new Promise<EmailConfirmationCTAMutationResponse>((done, reject) => {
+      commitMutation(environment, {
+        onCompleted: (data, errors) =>
+          errors && errors.length ? reject(errors) : done(data),
+        onError: error => reject(error),
+        mutation: graphql`
+          mutation EmailConfirmationCTAMutation {
+            sendConfirmationEmail(input: {}) {
+              confirmationOrError {
+                ... on SendConfirmationEmailMutationSuccess {
+                  unconfirmedEmail
+                }
+                ... on SendConfirmationEmailMutationFailure {
+                  mutationError {
+                    error
+                    message
+                  }
+                }
+              }
+            }
+          }
+        `,
+      })
+    }).then(({ sendConfirmationEmail: { confirmationOrError } }) => {
+      const emailToConfirm = confirmationOrError?.unconfirmedEmail
```

Comment:
In Eigen, we have this all wrapped within [a `try { } catch { }` ](https://github.com/artsy/eigen/blob/98b1ed6b737c323b325532553c1acf385f70777f/src/lib/Scenes/Home/Components/EmailConfirmationBanner.tsx#L54-L82), should we do the same here?

Reply:
i personally find the try/catch a little more difficult to reason about when used with async/await. I'm fine if we change this in the future though.